---
title: "ADVANCEMENT"
author: "Cletus Emmanuel"
date: '2022-07-22'
output:
  rmdformats::readthedown:
    toc_depth: 3
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```


```{r, echo=FALSE, message=FALSE, include=FALSE}
library(tidyverse)
library(readxl)
## library(magick)
# library(tcltk)
library(knitr)

```

## UYT.40.IB.MK.AG.ON.UB.OT

```{r, echo=FALSE, include=FALSE, message=FALSE, error=FALSE}
library(tidyverse)

# uyt40_sindex selection
setwd("~/Desktop/git_workspace/UYT40_sINDEX/")

uyt40_sindex <- read_excel("tables/BLUPS-UYT-40.xls") %>%
  janitor::clean_names() %>% 
  select(accession:index)
colnames(uyt40_sindex)[13] <- "sindex"

# list of checks present in dataset
checks <- c("IITA-TMS-IBA000070","TMEB419","TMS13F1160P0004","IITA-TMS-IBA30572","IITA-TMS-IBA980581")

# calculate for checks average 
checks_mean <- uyt40_sindex %>%
  filter(accession %in% checks) %>%
  add_row(accession = "check_mean", summarise(., across(where(is.numeric), mean))) %>%
  filter(accession == "check_mean")

# insert check mean into dataset
uyt40_sindex <- bind_rows(uyt40_sindex,checks_mean)

# dataset for percentage difference against checks average
uyt40_sindex_checkdiff <- uyt40_sindex %>% 
  select(-sindex) %>% 
  mutate(across(where(is.numeric), .fns = ~((./.[accession == "check_mean"]-1)*100))) %>% 
  mutate(sindex=uyt40_sindex$sindex) %>% 
  mutate(rank = factor(row_number()))


#  skimr::skim(uyt40_sindex)
# skimr::skim(uyt40_sindex_checkdiff)

# Hmisc::describe(uyt40_sindex)
# Hmisc::describe(uyt40_sindex_checkdiff)
# 
# psych::describe(uyt40_sindex)
# psych::describe(uyt40_sindex_checkdiff)
# 
# summarytools::descr(uyt40_sindex)
# summarytools::descr(uyt40_sindex_checkdiff)
# 
# summarytools::dfSummary(uyt40_sindex)
# summarytools::dfSummary(uyt40_sindex_checkdiff)


# pt_fun <- function(x, )
# 
# map2_df(checks_mean[,-1], uyt40_sindex[,-1], `/`) %>% 
#   -1 * 100 %>% 
#   add_column(accession = uyt40_sindex$accession, .before = "sprout")

```


```{r, include=TRUE, echo=FALSE, message=FALSE}
# barplot_checkdiff <- uyt40_sindex_checkdiff %>%
#   select(-c(rtwt)) %>%
#   select(accession,fyld,dyld,dm,shtwt,everything()) %>%
#   pivot_longer(-c(accession,rank,sindex), names_to = "traits", values_to = "values") %>%
# 
#  # group_by(traits) %>%
#   # top_frac(.1,sindex) %>%
#   #ungroup() %>%
# 
#   # mutate(traits = fct_reorder(traits, values)) %>%
#   ggplot(aes(fct_inorder(traits), values)) +
#   geom_col(aes(fill = values), stat = "identity", color=alpha("black",.3)) +
#   # facet_grid( vars(fct_inorder(traits)), vars(accession), scales = "free", space = "free") +
#   # geom_hline(aes(yintercept = 0, alpha = 0.1, color="blue")) +
#   geom_text(aes(label= round(values,2)), position = position_stack(vjust = 0.5)) +
#   # geom_text(aes(label=round(values, 2)), vjust=0) +
#   scale_fill_gradient2(low = "red", mid = "white", high = "green", midpoint= 0) +
#   scale_y_continuous(labels = function(x) paste0(x, "%")) +
#   facet_wrap(~accession, ncol = 1, strip.position = "right", scales = "free") +
#   labs(x = "traits", y = "percent difference from checks_mean") +
#   theme_bw() +
#   theme(
#   strip.background = element_rect(fill = "grey20", color = "grey80", size = 1),
#   strip.text = element_text(colour = "white", size = 5))
  # theme(strip.text.x = element_text(size = 8))
# ggsave(plot = barplot_checkdiff, "images/plots/SIvsTraits.jpg", width = 12, height = 50, limitsize = FALSE)
  
# ggsave("t10p_barplot1.pdf", width=20, height=20)
```
# Check Mean Data

### Table Data
<style> 
.table-wrapper {
  overflow: auto;
  height: 200px;
  width: 100%;
}
thead, th {
  top: 0;
  position: sticky;
  z-index: 2;
}
</style>


```{r, echo=FALSE, message=FALSE, error=FALSE}
library(knitr)
library(kableExtra)
uyt40_sindex %>% 
  mutate_if(is.numeric, round,2) %>% 
  kbl(.) %>%
  # add_header_above(c("a" = 5, "b" = 18)) %>%
  kable_paper() 

  #%>% scroll_box(width = "100%", height = "200px", fixed_thead = TRUE)
```
### Percentage comparism to check mean values

```{r, echo=FALSE, message=FALSE, error=FALSE}
uyt40_sindex_checkdiff %>% 
  mutate_if(is.numeric, round,2) %>% 
  kbl(.) %>%
  # add_header_above(c("a" = 5, "b" = 18)) %>%
  kable_paper()
  # scroll_box(width = "100%", height = "200px")

```



```{r, results='asis', include=FALSE,echo=FALSE, message=FALSE, error=FALSE}

plot_image_path <- "images/plots/SIvsTraits.jpg"
  cat("\n")
  cat("# Percentage difference against checks mean")
  cat("\n")
  cat("![](",plot_image_path,")")
  cat("<br> <br>")
```


# Correlation plots

```{r, echo=FALSE, message=FALSE,error=FALSE}
library(ggcorrplot)

corr <- round(cor(uyt40_sindex[,-1], use = "pairwise.complete.obs"), 1)
p.mat <- cor_pmat((uyt40_sindex[,-1]), use = "pairwise.complete.obs")

ggcorrplot(corr, method = c("square"), type = c("lower"), 
           ggtheme = ggplot2::theme_minimal, title = "",
           show.legend = TRUE, legend.title = "Correlation", show.diag = FALSE,
           colors = c("blue", "white", "red"), outline.color = "gray",
           hc.order = TRUE, hc.method = "complete", lab = TRUE,
           lab_col = "black", lab_size = 4, p.mat = NULL, sig.level = 0.05,
           insig = c("pch", "blank"), pch = 4, pch.col = "black",
           pch.cex = 5, tl.cex = 10, tl.col = "black", tl.srt = 45,
           digits = 2)


# install.packages("corrr")

# library(corrr)
# 
# column_to_rownames(uyt40_sindex, var =  "accession") %>%
#   corrr::correlate() %>%
#   corrr::focus(sindex) %>% 
#   dplyr::mutate(term = reorder(term, sindex)) %>%
#   ggplot2::ggplot(ggplot2::aes(term, sindex)) +
#   # color each bar based on the direction of the correlation
#   ggplot2::geom_col(ggplot2::aes(fill = sindex >= 0)) + 
#   ggplot2::coord_flip()

```

# Heatmaps 


```{r out.width="100%", fig.height=25, fig.width=15, fig.align='center', echo=FALSE, message=FALSE}
library(superheat)

u4h <- uyt40_sindex %>%
  # arrange(desc(sindex)) %>%
  column_to_rownames("accession")

dm <- paste(u4h$dm, "dm")
sindex <- paste(u4h$sindex, "sindex")

# set the text colors 
# identify all scaled values that fall below -0.3
ayt20.col <- scale(u4h) < -0.3
# set all values that satisfy the condition to "white"
ayt20.col <- gsub("TRUE", "white", ayt20.col)
# set all values that do not satisfy the condition to "black"
ayt20.col <- gsub("FALSE", "black", ayt20.col)
# convert to matrix
ayt20.col <- matrix(ayt20.col, ncol = ncol(u4h))

ayt20.size <- scale(u4h) + 1.2


set.seed(2016113)
superheat(u4h,
          # retain original order of rows/cols
          # pretty.order.rows = TRUE,
          # pretty.order.cols = TRUE,
          # scale the matrix columns
          scale = TRUE,
          # order the rows by selection index
          # order.rows = order(uyt40_sindex$sindex),
          
          # change the color
          # heat.col.scheme = "blue",
          # change the color (#b35806 = brown and #542788 = purple)
          heat.pal = c("red", "white", "darkgreen"),
          # Color transitions
          # heat.pal.values = c(0, 0.5, 1),
          # # color limits
          # heat.lim = c(-1, 2),
          # heat.na.col = "white", # na values
          # add row dendrogram
          row.dendrogram = TRUE,
          # add colun dendrogram
          col.dendrogram = TRUE,
          
          # clustering methods
          # clustering.method = "hierarchical",
          # generate column clusters
          # n.clusters.rows = 4,
          # left.label = 'variable'
          
          # cluster by sindex
          # membership.rows = fct_inorder(sindex)
          
          # plot title
          title = "Superheat for UYT 40 \n selection index",
          title.size = 5,
          # row title
          row.title = "accession",
          row.title.size = 5,
          # col title
          column.title = "traits",
          column.title.size = 5,
          
          # adjacent plots
          # # add selection index as a scatterplot next to the rows
          # yr = u4h$sindex,
          # yr.axis.name = "selection index"
          
          # add text matrix
          # X.text = round(as.matrix(u4h), 1),
          # X.text.col = ayt20.col,
          # #X.text.size = 4,
          # X.text.size = ayt20.size,
          # X.text.angle = 12,
          
          # change the size of the labels
          left.label.size = 0.24,
          bottom.label.size = 0.24,
          
          # change the size of the label text
          left.label.text.size = 4,
          bottom.label.text.size = 8,
          
          # # change the color of the labels
          # left.label.col = "white",
          # bottom.label.col = c("#b3e2cd","#fdcdac","#e5d8bd"),
          
          # change the color of the label text
          left.label.text.col = "black",
          bottom.label.text.col = "black",
          
          # change the angle of the label text
          bottom.label.text.angle = 90,
          left.label.text.alignment = "center",
          bottom.label.text.alignment = "center",
          
          # # remove the grid
          # grid.hline = FALSE,
          # grid.vline = FALSE,
          
          # # change the grid color and size
          # grid.hline.col = "white",
          # grid.vline.col = "white",
          # grid.hline.size = 2,
          # grid.vline.size = 2,
          
          # # cluster the heatmap
          # n.clusters.rows = 3,
          # left.label = "variable",
          # n.clusters.cols = 2,
          # bottom.label = "variable"
          
          # # remove the legend
          # legend = FALSE,
          
          #  # make the legend bigger
          # legend.height = 0.5,
          # legend.width = 2,
          # legend.text.size = 20,
          
          # # cluster by gears
          # membership.rows = sindex,
          # 
          # # place each variable in its own cluster
          # membership.cols = 1:ncol(u4h),
          # bottom.label = "variable",
          # 
          # # smooth the heatmap within clusters
          # smooth.heat = TRUE
          )
          



# plot a super heatmap


```

```{r out.width="100%", include=FALSE, echo=FALSE, message=FALSE}
# extracting the clusters

library(superheat)
superheatmap <- superheat(u4h,
                          # scale the matrix columns
                          scale = TRUE,
                          # generate three column clusters
                          n.clusters.rows = 3,
                          left.label = 'variable',
                          print.plot = F)

# extract the clusters
superheatmap$membership.rows


superheat(dplyr::select(u4h, -sindex, -fyld), 
          # scale the variables/columns
          scale = T,
          
          # cluster the rows
          membership.rows = paste(u4h$fyld, "fyld"),
          left.label = "variable",
          
          # change the size of the label text
          left.label.text.size = 2.2,
          bottom.label.text.size = 4,
          
          # add selection index as a scatterplot next to the rows
          yr = u4h$sindex,
          yr.axis.name = "selection index",
          yr.obs.col = rep("paleturquoise4", nrow(u4h)),
          yr.point.size = 4
          )

# set a color vector

```

```{r out.width="100%", fig.height= 25, fig.width= 15, echo=FALSE, message=FALSE, include=FALSE}
print_col_checks <- u4h %>% 
  mutate(row_number = row_number()) %>% 
 .[checks,] #%>%
 #  select(row_number) %>% 
 #  as.vector() 


point.col <- rep("wheat3", nrow(u4h))
# color checks
point.col[print_col_checks[c(1:ncol(print_col_checks)),13]] <- "red"

# the side plot here highlights the checks in red colour 
# plot a super heatmap
superheat(dplyr::select(u4h, -sindex), 
          # scale the variables/columns
          scale = T,
          # order the rows by selection index
          order.rows = order(uyt40_sindex$sindex),
          
          # change the size of the label text
          left.label.text.size = 4,
          bottom.label.text.size = 4,

          # add selection index as a scatterplot next to the rows
          yr = u4h$sindex,
          yr.axis.name = "selection index",
          # change the color of the points
          yr.obs.col = point.col,
          yr.point.size = 4)

```

# Heat Maps with Correlation Barplot

```{r, echo=FALSE, message=FALSE, fig.align='center', fig.height=25, fig.width=15, out.width="100%"}
superheat(dplyr::select(u4h, -sindex), 
          # scale the variables/columns
          scale = T,
          # order the rows by selection index
          order.rows = order(uyt40_sindex$sindex),

          # # add selection index as a scatterplot next to the rows
          # yr = u4h$sindex,
          # yr.axis.name = "selection index",
          # # change the color of the points
          # yr.obs.col = point.col,
          # yr.point.size = 4
          
          # # add selection index as a line plot next to the rows
          # yr =u4h$sindex,
          # yr.axis.name = "selection index",
          # yr.plot.type = "line",
          # # order the rows by mpg
          # order.rows = order(u4h$sindex)
          
          # # loess curve
          # # add selection index as a smoothed line plot next to the rows
          # yr = u4h$sindex,
          # yr.axis.name = "selection index",
          # yr.plot.type = "smooth",
          # # change the line thickness and color
          # yr.line.size = 4,
          # yr.line.col = "red4",
          # # order the rows by b
          # order.rows = order(u4h$b)
          
          # # linear regression line
          # # add selection index as a smoothed line plot  next to the rows
          # yr = u4h$sindex,
          # yr.axis.name = "selection index",
          # yr.plot.type = "smooth",
          # smoothing.method = "lm",
          # # change the line thickness and color
          # yr.line.size = 4,
          # yr.line.col = "plum4",
          # # order the rows by b
          # order.rows = order(u4h$b)
          
          # # scatterplot with connecting line plot
          # # add selection index as a scatter line plot next to the rows
          # yr = u4h$sindex,
          # yr.axis.name = "selection index",
          # yr.plot.type = "scatterline",
          # # change the line color
          # yr.line.col = "tomato3",
          # yr.obs.col = rep("orange", nrow(u4h)),
          # yr.point.size = 4,
          # # order the rows by b
          # order.rows = order(u4h$b)
          
          # # scatterplot with smooth line
          # # add selection index as a scatter smoothed plot next to the rows
          # yr = u4h$sindex,
          # yr.axis.name = "selection index",
          # yr.plot.type = "scattersmooth",
          # # change the line color
          # yr.line.col = "tomato3",
          # yr.obs.col = rep("orange", nrow(u4h)),
          # # order the rows by b
          # order.rows = order(u4h$b)
          
          # # add selection index as a barplot next to the rows
          # yr = u4h$sindex,
          # yr.axis.name = "selection index",
          # yr.plot.type = "bar",
          # # set bar colors
          # yr.bar.col = "black",
          # yr.cluster.col = c("beige", "white", "beige")
          
          # change the size of the label text
          left.label.text.size = 6,
          bottom.label.text.size = 6,
          
          # change the size of the labels
          left.label.size = 0.3,
          # bottom.label.size = 0.24,
          
          # add selection index as a scatterplot next to the rows
          yr = u4h$sindex,
          yr.axis.name = "selection index",
          yr.plot.size = 0.2,
          # yr.lim = c(0, 60),
          # change the color of the points
          yr.obs.col = point.col,
          yr.point.size = 4,
          # add correlation between each variable and selection index
          yt = cor(u4h)[-12,"sindex"],
          yt.plot.type = "bar",
          yt.axis.name = "Correlation with \n selection index",
          #yt.lim = c(-1.5, 1)
          # yt.axis.size = 14,
          yt.axis.name.size = 8,
          yt.plot.size = 0.35
          )




```




```{r out.width="100%", echo=FALSE, message=FALSE, error=FALSE, include=FALSE}
# plot a super heatmap

# library(dplyr)
# sindex.per.cluster <- u4h %>% 
#   group_by(b) %>% 
#   summarize(sindex.avg = mean(sindex)) %>% 
#   select(sindex.avg) %>%
#   unlist
# superheat(dplyr::select(u4h, -sindex), 
#           # scale the variables/columns
#           scale = T,
#           
#           # cluster the rows
#           membership.rows = paste(u4h$b, "b"),
#           left.label = "variable",
#           
#           # change the size of the label text
#           left.label.text.size = 3,
#           bottom.label.text.size = 4,
#           
#           # change the size of the labels
#           left.label.size = 0.3,
#           # bottom.label.size = 0.24,
#           
#           # boxplot
#           # add sindex per cluster as a boxplot
#           yr = u4h$sindex,
#           yr.axis.name = "selection index",
#           yr.plot.type = "boxplot")
# 
# # ayt20s_cor <- cor(uyt40_sindex[-1])
# # 
# # superheat::superheat(ayt20s_cor,
# #                      # scale = TRUE,
# #                      row.dendrogram = TRUE,
# #                      col.dendrogram = TRUE,
# #                      
# #                      # add text matrix
# #                      X.text = round(as.matrix(ayt20s_cor), 1),
# #                      X.text.col = "white",
# #                      
# #                      # change the size of the label text
# #                      left.label.text.size = 3,
# #                      bottom.label.text.size = 3)
# 
# 
# # # best way to save our superheat plot
# # 
# # png("superheat.png", height = 900, width = 800)
# # superheat(X = mtcars, scale = T)
# # dev.off()

```


```{r, echo=FALSE, message=FALSE}
###trial <- read_excel(file.choose(), sheet = "Advancement")

trial <- uyt40_sindex
###attach(trial)

# tria <- trial %>% select(-c("Selected"))

get_genotypes <- function(){
  return(trial$accession)
}
```


```{r, echo=FALSE, message=FALSE}
####image_list <- list.files(path = tclvalue(tkchooseDirectory()))

### Select a directory and import images
image_list <- list.files(path = "images/photos/", pattern = ".jpg")
##image_list
##head(image_list)
```



```{r, echo=FALSE, message=FALSE}

get_images <- function(genotype, pattern){
list_gen <- c()
  for(i in 1:length(image_list)){
    if(stringr::str_detect(image_list[[i]], regex(genotype)) == TRUE){
      if(stringr::str_detect(image_list[[i]], regex(pattern)) == TRUE){
        list_gen[[length(list_gen) + 1]] <- image_list[[i]]
      }
    }
  }
  return(list_gen)
}

# get_images(get_genotypes()[[1]], "rep1")
# get_images(get_genotypes()[[1]], "rep2")
```


```{r, echo=FALSE, message=FALSE}

table_chunk <- function(gene){
  df <- trial %>% filter(accession == gene)
  FYLD <- df$fyld
  DYLD <- df$dyld
  DM <- df$dm
  SI <- df$sindex
  new_df <- data.frame(Trait = c("FYLD", "DYLD", "DM", "SINDEX"), 
                       Value = c(FYLD, DYLD, DM, SI))
  return(new_df)
}

# table_chunk(get_genotypes()[[1]])

```


```{r, echo=FALSE, message=FALSE, include=FALSE}
list_of_plots <- list.files(path = "images/plots/barplots/")
list_of_plots

get_plot <- function(genotype){
# list_gen <- c()
  plot <- ""
  for(i in 1:length(list_of_plots)){
    if(stringr::str_detect(list_of_plots[[i]], regex(genotype)) == TRUE){
        plot <- list_of_plots[i]
    }
  }
  return(plot)
}

```

# Genotype Images

```{r, results="asis", echo=FALSE, message=FALSE}
  genotype <- get_genotypes()
  genotypes <- get_genotypes()
  i <- 1
  cat("\n")
  for(i in 1:length(genotypes)){
    j <- 1
    get_images(genotypes[[i]], "rep1")
    if(!length(get_images(genotypes[[i]], "rep1")) < 1 || !length(get_images(genotypes[[i]], "rep2")) < 1 || 
       !length(get_plot(genotypes[[i]])) < 1){
      rep1_images <- get_images(genotypes[[i]], "rep1")
      rep2_images <- get_images(genotypes[[i]], "rep2")
      cat("\n")
      cat("### ", genotypes[[i]])
      cat("\n")
      df <- table_chunk(genotypes[[i]])
      print(kable(df, align = "l", caption = ""))
      cat("<br> \n")
      plot_image <- get_plot(genotype = genotypes[[i]])
      plot.path <- paste0("images/plots/barplots/",plot_image)
      cat("![](",plot.path,") <br>")
      cat("<center>",genotypes[[i]], "Against check mean </center> <br> <br>")
      cat("\n")
      if(!length(get_images(genotypes[[i]], "rep1")) < 1) {
        for(j in 1:length(rep1_images)){
          cat("\n")
          plot
          image.path <- paste0("images/photos/",rep1_images[[1]])
          cat("![](",image.path,") <br>")
          cat("<center> REP 1 </center>")
        }
      }
      if(!length(get_images(genotypes[[i]], "rep2")) < 1){
        for(j in 1:length(rep2_images)){
          image.path2 <- paste0("images/photos/",rep2_images[[1]])
          cat("![](",image.path2,") <br>")
          cat("<center> REP 2 </center> <br> <br>")
          cat("\n")
        }
      }
    }
  }

#   library(cowplot)
#   lapply(1:length(genotype), function(i){
#   
#   rep1 <- get_images(genotype = genotype[[i]], "rep1")
#   rep2 <- get_images(genotype = genotype[[i]], "rep2")
#   
#   if(!is.null(rep1) || !is.null(rep2)){
#     reps <- append(rep1, rep2)
#     img <- c()
#     for(j in 1:length(reps)){
#       path_ <- paste0("images/21.GS.C4B.AYT.20.IB_resize/",reps[[j]])  
#       img[[length(img) + 1]] <- path_
#     }
#     
#     length(img)
#     
#     cat("\n")
#     cat("### ", genotype[[i]])
#     cat("\n")
#     cat(kable(table_chunk(genotype[[i]]), align = "l"))
#     p1 <- ggdraw() + draw_image(img[[1]], scale = 0.5)
#     if(length(img) > 1){
#       p2 <- ggdraw() + draw_image(img[[2]], scale = 0.5)
#       x <- plot_grid(p1,p2, labels = c("Rep 1","Rep 2")))
#       cat("\n")
#     } else if(length(img) == 1){
#       if(str_detect(reps[[1]], "rep1")){
#         x <- plot_grid(p1, labels = c("Rep 1"))  
#         cat("\n")
#       } else {
#         x <- plot_grid(p1, labels = c("Rep 2"))
#         cat("\n")
#       }
#     }
#   }
# }) -> suppress
  
```


# Relationship between Traits and SINDEX

```{r, results='asis', echo=FALSE, message=FALSE,error=FALSE}
# uyt40s_long <- uyt40_sindex %>%
#   select(-c(rtwt)) %>%
#   select(accession,fyld,dyld,dm,shtwt,everything()) %>%
#   pivot_longer(-c(accession,sindex), names_to = "traits", values_to = "values")
# 
# library(ggrepel)
# 
# uyt40s_long %>%
# ggplot(aes(values, sindex)) +
# #geom_hex(aes(color = SI_W), bins = 30) +
# geom_point(aes(color = sindex > 80), alpha = 0.8) +
# #geom_hex(aes(color = SI_W), alpha = 0.6) +
# # scale_colour_gradient(low = "blue", high = "red") +
# facet_wrap(~traits, scales = "free", ncol = 2) +
# geom_text_repel(data=subset(uyt40s_long, sindex > 80), aes(values, sindex,label=accession), size = 4)
# #
# ggsave("images/plots/SIvsTraits.jpg", width = 12, height = 15)
# cat("![](","images/plots/SIvsTraits.jpg",")")
```

```{r, results='asis', echo=FALSE, message=FALSE,error=FALSE}
# ayt20_scd_long <- uyt40_sindex_checkdiff %>%
#   select(-c(l,a,b, rtwt)) %>%
#   select(accession,fyld,dyld,dm,shtwt,everything()) %>%
#   pivot_longer(-c(accession,rank,sindex), names_to = "traits", values_to = "values")
# 
# library(ggrepel)
# 
# ayt20_scd_long %>%
#   ggplot(aes(values, sindex)) +
#   #geom_hex(aes(color = SI_W), bins = 30) +
#   geom_point(aes(color = sindex > 80), alpha = 0.8) +
#   #geom_hex(aes(color = SI_W), alpha = 0.6) +
#   # scale_colour_gradient(low = "blue", high = "red") +
#   facet_wrap(~traits, scales = "free", ncol = 2) +
#   geom_text_repel(data=subset(ayt20_scd_long, sindex > 80),
#                   aes(values, sindex,label=accession), size = 4)
#   ggsave("../RMKRDpRCT/images/plots/SIvsTraits_p.jpg", width = 12, height = 15)

  # cat("![](","images/plots/SIvsTraits_p.jpg",")")

```

# Phenotypic Values

```{r, echo=FALSE, message=FALSE, error=FALSE}
uyt40_sindex %>% 
  pivot_longer(-accession, names_to = "traits", values_to = "values") %>% 
  ggplot(aes(traits, values, fill = traits)) +
  geom_violin(alpha = 0.7, color=NA) +
  geom_jitter(alpha = 0.3, width = 0.1) +
  facet_wrap(~traits, scales = "free") +
  theme_bw() +
  theme(legend.position="none") +
  labs(title = "Distribution of Raw Phenotypic Values and Selection Index")
```

```{r, echo=FALSE, message=FALSE, error=FALSE}
uyt40_sindex_checkdiff %>% 
  select(-c(sindex)) %>% 
  pivot_longer(-c(accession,rank), names_to = "traits", values_to = "values") %>%
   ggplot(aes(traits, values, fill = traits)) +
  geom_violin(alpha = 0.7, color=NA) +
  geom_jitter(alpha = 0.3, width = 0.1) +
  scale_y_continuous(labels = function(x) paste0(x, "%")) +
  facet_wrap(~traits, scales = "free") +
  theme_bw() +
  theme(legend.position="none") +
  labs(title = "Distribution of Raw Phenotypic Values Difference from Check Average (%)")

```





